docs/blog/2023/07/high-availability-for-klaw.rst

```diff
@@ -0,0 +1,274 @@
+- **IP-hash:** A hash-function based on the client's IP address is used to determine the server to which the next request should be sent.
+
+Detailed information can be found on the official Nginx documentation.
+
+Database authentication in Klaw
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In Klaw, you can configure database-level authentication by setting `klaw.login.authentication.type: db` in the core mode application properties.
+
+With database authentication, Klaw uses the Spring JSESSION ID. When deploying Klaw in HA mode across more than one instance to ensure uninterrupted user access, it's advisable to configure the IP-Hash load balancing method.
+
+Below is an example of Nginx configuration using the IP-Hash method:
+
+**Nginx configuration**
+::
+      upstream klawgcp {
+	  ip_hash; // load balancing method
+      server klawhost:9097;
+	  server klawhost:9097;
+        }
+        server {
+                        listen 80;
+                        listen [::]:80;
+                        ssl on;
+
+                        ssl_certificate     /opt/klaw/certs/service.cert;
+                        ssl_certificate_key /opt/klaw/certs/service.key;
+                        ssl_protocols TLSv1.2;
+                        server_name serverhost;
+                        location / {
+                                proxy_pass https://klawgcp;
+                            }
+         }
+
+
+Using IP-Hash method, sessions are maintained by tracking the client's IP address.
+
+Single Sign-On (SSO) authentication in Klaw
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For SSO authentication, configure Klaw by setting `klaw.login.authentication.type: ad` in the core mode application properties. When SSO is enabled, either Round-Robin or Least-Connected load balancing methods can be used.
+
+Below is an example Nginx configuration using Round-Robin load balancing:
+
+**Nginx configuration**
+
+::
+
+    upstream klawgcp {
+        round-robin; // load balancing method
+        server klawhost:9097;
+        server klawhost:9097;
+    }
+    server {
+            listen 80;
+            listen [::]:80;
+            ssl on;
+
+            ssl_certificate     /opt/klaw/certs/service.cert;
+            ssl_certificate_key /opt/klaw/certs/service.key;
+            ssl_protocols TLSv1.2;
+            server_name serverhost;
+            location / {
+                    proxy_pass https://klawgcp;
+                }
+    }
+
+
+
+Klaw deployment model in high availability (HA) mode
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In the high availability mode, Nginx routes the requests to Klaw instances which are connected to a common data store such as Postgres. Each Klaw instance comprises both the Governance and Cluster API applications. You may choose to deploy these applications on separate machines for increased robustness.
+Below is the deployment model of Klaw in HA mode.
+
+
+Klaw in HA mode with Nginx Load balancer
```

Comment:
@muralibasani Just realized that there is an image missing here. Apart from this, the rest looks good.

Reply:
@harshini-rangaswamy thank you. Added the image now.